c/registerProduct.c

```diff
@@ -0,0 +1,200 @@
+
+
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+
+  SPDX-License-Identifier: EPL-2.0
+
+  Copyright Contributors to the Zowe Project.
+*/
+
+
+#ifdef METTLE
+#include <metal/metal.h>
+#include <metal/stddef.h>
+#include <metal/stdio.h>
+#include <metal/stdlib.h>
+#include <metal/string.h>
+#include <metal/stdarg.h>
+#include "metalio.h"
+
+#else
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stdarg.h>
+#include <sys/stat.h>
+#include <iconv.h>
+#include <dirent.h>
+#include <errno.h>
+#include <pthread.h>
+#include <signal.h>
+#include <assert.h>
+#include <stdbool.h>
+
+#endif
+
+#include "zowetypes.h"
+#include "alloc.h"
+#include "utils.h"
+#ifdef __ZOWE_OS_ZOS
+#include "zos.h"
+#endif
+#include "bpxnet.h"
+#include "collections.h"
+#include "unixfile.h"
+#include "socketmgmt.h"
+#include "le.h"
+#include "logging.h"
+#include "scheduling.h"
+#include "json.h"
+
+#include "xml.h"
+#include "httpserver.h"
+#include "httpfileservice.h"
+#include "dataservice.h"
+#include "envService.h"
+#ifdef __ZOWE_OS_ZOS
+#include "zosDiscovery.h"
+#endif
+#include "charsets.h"
+#include "plugins.h"
+#ifdef __ZOWE_OS_ZOS
+#include "datasetjson.h"
+#include "authService.h"
+#include "securityService.h"
+#include "zis/client.h"
+#endif
+
+#include "zssLogging.h"
+#include "serviceUtils.h"
+#include "unixFileService.h"
+#include "omvsService.h"
+#include "datasetService.h"
+#include "serverStatusService.h"
+#include "rasService.h"
+#include "certificateService.h"
+#include "registerProduct.h"
+
+#include "jwt.h"
+
+#define PRODUCT "ZLUX"
+#ifndef PRODUCT_MAJOR_VERSION
+#define PRODUCT_MAJOR_VERSION 0
+#endif
+#ifndef PRODUCT_MINOR_VERSION
+#define PRODUCT_MINOR_VERSION 0
+#endif
+#ifndef PRODUCT_REVISION
+#define PRODUCT_REVISION 0
+#endif
+#ifndef PRODUCT_VERSION_DATE_STAMP
+#define PRODUCT_VERSION_DATE_STAMP 0
+#endif
+
+unsigned long long __registerProduct(const char *major_version,
+                                const char *pid,
+                                const char *product_owner,
+                                const char *product_name,
+                                const char *feature_name){
+
+  /* Creates buffers for registration product info */
+
+  char str_product_owner[17];
+  char str_feature_name[17];
+  char str_product_name[17];
+  char str_pid[9];
+  char version[9];
+  unsigned long long ifausage_rc = 0;
+  IFAARGS_t *arg;
+
+  /* Check if SMF is Active first  */
+  char *xx = ((char *__ptr32 *__ptr32 *)0)[4][49];
```

Comment:
This location (in CVT) decimal offset 196 (4*49) is CVTSMCA.

```ADDRESS OF THE SYSTEM MANAGEMENT CONTROL AREA (SMCA) IF THE SYSTEM MANAGEMENT FACILITIES (SMF) OPTION IS PRESENT IN THE SYSTEM. OTHERWISE, ZERO.
```
I also found the CVT tag in deps/zowe-common-c/h/zos.h.

I will provide and use #define in here.